### ISSUE TITLE
Implement RFC 7591 and 7592 (dynamic client registration)

### ISSUE THREAD

**Issue by IvanAnishchuk** [jazzband/django-oauth-toolkit#670]
Basically, in a project I'm working on that uses oauth-toolkit there's a need for api-based device registration and since there's a standard for that it seems to make sense to try and implement it here in a reusable manner.

https://tools.ietf.org/html/rfc7591
https://tools.ietf.org/html/rfc7592

Did anybody attempt anything like this before, does anybody have a similar need or any ideas regarding implementation? Any guidance or advice otherwise?

Basically, I'm prepared to do all the work myself but I'd prefer to make it generic and reusable as it feels like a nice feature for everyone.

**Comment by JonathanHuot** [comment]
Hi @IvanAnishchuk, the django-oauth-toolkit project is using https://github.com/oauthlib/oauthlib library where it contains all the RFC code.

I'll suggest to do in parallel the implementation of RFC7591 in oauthlib and also the oauthlib's wrapper in django-oauth-toolkit. It gives the best separation of concerns between the web framework and the protocol specification.

On oauthlib's side, we will be happy to review any PR and release new versions once integrated.

**Comment by lullis** [comment]
I've looked into both this repository and oauthlib and it seems that no significant work has been done on this so far, and I am tasked to implement it for [Aleksis](https://edugit.org/AlekSIS). Any recommendation / suggestion that you could give before I dive into coding?

**Comment by lullis** [comment]
@JonathanHuot, can you please chime on this? 

Unless I am missing something, all that we need to implement this on Django is an analog to `oauth2_provider.views.application.ApplicationRegistration`, but that can handle JSON instead of being a FormView. No integration with OAuthLib is needed.

**Comment by zacharypodbela** [comment]
I opened a PR for this #1667
